src/utils/cache.h

```diff
@@ -62,6 +62,8 @@ class LruCache {
   // If @pinned, pins inserted element, Unpin has to be called to unpin.
   // In any case, puts element to front of the queue (makes it last to evict).
   V* Insert(K key, std::unique_ptr<V> val, bool pinned = false) {
+    if (capacity_ == 0) return val.get();
```

Comment:
std::unique_ptr will be destroyed immediately on exit, so the function will return hanging pointer.

Reply:
On a second look, this function is never used with pinning.

With that, that functionality can be removed (part with pinned==true), and this function should return void instead of V*.
Then it seems correct.

Reply:
After this change there is no need for the `pins` argument in the `Item` constructor.